CMakeLists.txt

```diff
@@ -15,14 +15,21 @@ option(BUILD_TEST    "Build tests" ON)
 option(BUILD_XTT     "Build with XTT support" ON)
 option(BUILD_KEYGEN  "Build with keygen tool" ON)
 option(CHECK_FORMAT  "Check source format during build" ON)
-
+option(USE_TSS_ENGINE  "Use tpm2-tss-engine" OFF)
 
 include(GNUInstallDirs)
 
 find_package(LibConfig REQUIRED QUIET 1.5)
 find_package(LibUV REQUIRED QUIET 1.9)
 find_package(OpenSSL REQUIRED QUIET 1.0)
 
+if(USE_TSS_ENGINE)
```

Comment:
The intent behind this is good: try to prevent the conflict between the tpm2-software and xaptum-tpm TSS that you ran into yesterday.   But I think this is not the best approach for two reasons:

1. There's no actual compile-time dependency on the TSS engine. Using it is strictly a runtime decision.  So we don't need to detect a conflict at compile-time.  It's fine to build our `tls_tpm.c` code even if XTT is enabled; it's only a problem if someone tries to load a TSS key.
1. The XTT suite needs to be updated to support the tpm2-software TSS, so these checks won't be needed anyway.  This approach adds a bunch of changes (to the `CMakeLists.txt` and many `ifdef`s) that would just need to be removed.

Instead, let's do it at runtime in `enftun_tls_load_credentials()` based on the value of `BUILD_XTT`. Something like:

```
    if (!key_used && enftun_tls_tpm_is_tpm_key(key_file))
    {
#ifndef BUILD_XTT
        key_used = enftun_tls_tpm_use_key(tls, key_file);
        if (!key_used)
        {
            enftun_log_info("Failed to load client TLS key from TPM %s:",
                            key_file);
        }
#else 
        enftun_log_error("TPM key not supported when XTT is enabled.");
#endif
    }
```

Once the XTT suite is updated to support the tpm2-software TSS, we'll only have to remove that one `ifdef`.

Reply:
Done